internal/services/copilot_studio_application_insights/api_copilot_studio_application_insights.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package copilot_studio_application_insights
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/microsoft/terraform-provider-power-platform/internal/api"
+	"github.com/microsoft/terraform-provider-power-platform/internal/constants"
+)
+
+func newCopilotStudioClient(apiClient *api.Client) client {
+	return client{
+		Api: apiClient,
+	}
+}
+
+type client struct {
+	Api *api.Client
+}
+
+func (client *client) getCopilotStudioEndpoint(ctx context.Context, environmentId string) (string, error) {
+	env, err := client.getEnvironment(ctx, environmentId)
+	if err != nil {
+		return "", err
+	}
+	if env.Properties.RuntimeEndpoints.PowerVirtualAgents == "" {
```

Comment:
might want to do some more robust checking to ensure the structure you're assuming is there is actually there
```
if env == nil || env.Properties == nil || env.Properties.RuntimeEndpoints == nil {
return "", fmt.Errorf("invalid environment or missing properties")
}
```